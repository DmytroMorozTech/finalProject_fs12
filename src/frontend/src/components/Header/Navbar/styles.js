import {makeStyles} from '@material-ui/core/styles'

export default makeStyles(() => ({

  navbar: {
    display: 'flex',
    alignItems: 'center',
    '& > div': {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      marginRight: '40px'
    }
  }
}))