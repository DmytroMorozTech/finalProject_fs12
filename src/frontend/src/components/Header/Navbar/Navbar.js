import Item from './Item/Item'
import styles from './styles'
import HomeRoundedIcon from '@material-ui/icons/HomeRounded'
import SupervisorAccountRoundedIcon from '@material-ui/icons/SupervisorAccountRounded'
import BusinessCenterRoundedIcon from '@material-ui/icons/BusinessCenterRounded'
import SmsRoundedIcon from '@material-ui/icons/SmsRounded'
import NotificationsRoundedIcon from '@material-ui/icons/NotificationsRounded'
import AccountCircleRoundedIcon from '@material-ui/icons/AccountCircleRounded'
import React from 'react'
import { Hidden } from '@material-ui/core'

function Navbar () {
  const classes = styles()

  const items = [
    { Icon: <HomeRoundedIcon fontSize='inherit'/>, title: 'Home', arrow: false, toggleMenu: false, to: '/home' },
    { Icon: <SupervisorAccountRoundedIcon fontSize='inherit'/>, title: 'Network', arrow: false, toggleMenu: false, to: '/network' },
    { Icon: <BusinessCenterRoundedIcon fontSize='inherit'/>, title: 'Jobs', arrow: false, toggleMenu: false, to: '/jobs' },
    { Icon: <SmsRoundedIcon fontSize='inherit'/>, title: 'Messages', arrow: false, toggleMenu: false, to: '/messages' },
    { Icon: <NotificationsRoundedIcon fontSize='inherit'/>, title: 'Notifications', arrow: false, toggleMenu: false, to: '/notifications' },
    { Icon: <AccountCircleRoundedIcon fontSize='inherit'/>, title: 'Me', arrow: true, toggleMenu: true, to: '/personal' }
  ]

  return (
    <div className={classes.navbar}>
      {items.map(({ Icon, title, arrow, toggleMenu, onClick, to }, i) => (
        <Item key={i} Icon={Icon} title={title} arrow={arrow} toggleMenu={toggleMenu} to={to} />
      ))}
      <Hidden mdDown>
        <span className={classes.margin}> </span>
      </Hidden>
    </div>
  )
}

export default Navbar
