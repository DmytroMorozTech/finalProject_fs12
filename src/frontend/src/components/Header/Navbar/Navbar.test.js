import { shallow } from 'enzyme'
import * as redux from 'react-redux'
import React from 'react'
import Navbar from './Navbar'

const setUp = (props) => shallow(<Navbar {...props}/>)

describe('should render Navbar component', () => {
  let spyOnUseSelector
  let spyOnUseDispatch
  let mockDispatch

  let component

  beforeEach(() => {
    spyOnUseSelector = jest.spyOn(redux, 'useSelector')
    spyOnUseSelector.mockReturnValue(false)
    spyOnUseDispatch = jest.spyOn(redux, 'useDispatch')
    mockDispatch = jest.fn()
    spyOnUseDispatch.mockReturnValue(mockDispatch)

    component = setUp()
  })
  afterEach(() => {
    jest.restoreAllMocks()
  })
  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
  it('should pass values to NavbarItem', () => {
    expect.stringMatching('Home')
    expect.stringMatching('Network')
    expect.stringMatching('Jobs')
    expect.stringMatching('Messages')
    expect.stringMatching('Notifications')
    expect.stringMatching('Me')
  })
})
