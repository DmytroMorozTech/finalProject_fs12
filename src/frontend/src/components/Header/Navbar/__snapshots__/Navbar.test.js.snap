// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Navbar component to match snapshot 1`] = `
<div
  className="makeStyles-navbar-1"
>
  <NavbarItem
    Icon={
      <Memo
        fontSize="inherit"
        onClick={[Function]}
      />
    }
    arrow={false}
    badge={false}
    key="0"
    title="Home"
    to="/home"
    toggleMenu={false}
  />
  <NavbarItem
    Icon={
      <Memo
        fontSize="inherit"
        onClick={[Function]}
      />
    }
    arrow={false}
    badge={false}
    exact={false}
    key="1"
    title="Network"
    to="/network"
    toggleMenu={false}
  />
  <NavbarItem
    Icon={
      <Memo
        fontSize="inherit"
        onClick={[Function]}
      />
    }
    arrow={false}
    badge={false}
    exact={false}
    key="2"
    title="Jobs"
    to="/jobs"
    toggleMenu={false}
  />
  <NavbarItem
    Icon={
      <Memo
        fontSize="inherit"
        onClick={[Function]}
      />
    }
    arrow={false}
    badge={true}
    key="3"
    title="Messages"
    to="/messages"
    toggleMenu={false}
  />
  <NavbarItem
    Icon={
      <Memo
        fontSize="inherit"
        onClick={[Function]}
      />
    }
    arrow={false}
    badge={false}
    key="4"
    title="Notifications"
    to="/notifications"
    toggleMenu={false}
  />
  <NavbarItem
    Icon={
      <Memo
        fontSize="inherit"
        onClick={[Function]}
      />
    }
    arrow={true}
    badge={false}
    key="5"
    title="Me"
    to="/profiles/undefined"
    toggleMenu={true}
  />
</div>
`;
